odoo/addons/base/models/res_company.py

```diff
@@ -309,6 +309,14 @@ def create(self, vals_list):
         # Make sure that the selected currencies are enabled
         companies.currency_id.sudo().filtered(lambda c: not c.active).active = True
 
+        companies_needs_l10n = (
+            not tools.config['test_enable']
+            and (self.env.registry.ready or not self.env.registry._init)
```

Comment:
Put these conditions in `install_l10n_modules` to [Keep it DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)